packages/redux-routine/src/test/is-generator.js

```diff
@@ -4,17 +4,16 @@
 import isGenerator from '../is-generator';
 
 describe( 'isGenerator', () => {
-	it( 'should return false if not a generator', () => {
-		[
-			undefined,
-			null,
-			10,
-			'foo',
-			function() {},
-			function* () {},
-		].forEach( ( value ) => {
-			expect( isGenerator( value ) ).toBe( false );
-		} );
+	test.each( [
+		[ undefined ],
+		[ null ],
+		[ 10 ],
+		[ 'foo' ],
+		[ [ 0, 1, 2, 3 ] ],
+		[ function() {} ],
+		[ function*() {} ],
+	] )( 'should return false if %o', ( value ) => {
```

Comment:
The output of this can be a bit awkward, notably for the functions:

```
PASS packages/redux-routine/src/test/is-generator.js
  isGenerator
    ✓ should return false if undefined (1ms)
    ✓ should return false if null (1ms)
    ✓ should return false if 10 (1ms)
    ✓ should return false if 'foo'
    ✓ should return false if [ 0, 1, 2, 3, [length]: 4 ] (1ms)
    ✓ should return false if [Function] {
  [length]: 0,
  [name]: '',
  [prototype]: { [constructor]: [Circular] }
} (1ms)
    ✓ should return false if [GeneratorFunction] {
  [length]: 0,
  [name]: '',
  [prototype]: Object [Generator] {}
}
    ✓ should return false if an imposter! (1ms)
    ✓ should return false if an async generator (1ms)
    ✓ should return true if a generator
```

Granted, I like that it improves the output by creating separate cases for each item under test (improved debuggability).

A few options I might suggest to consider:

- Revert to the previous implementation
- Include a second (optional?) member of the array, to use as the "label" for the test ([example](https://github.com/WordPress/gutenberg/blob/17e5c2d870d84fb6de48dcd5bc3c38cd0c0fb0d0/packages/api-fetch/src/middlewares/test/preloading.js#L80-L83))
- Leave it as it is and consider the output fine enough

Reply:
I adjusted the output a bit:

- name the functions to differentiate
- use `%p` pretty format

I think that leaves a good result:

```
 PASS  packages/redux-routine/src/test/is-generator.js
  isGenerator
    ✓ should return false for undefined (2ms)
    ✓ should return false for null (1ms)
    ✓ should return false for 10
    ✓ should return false for "foo" (1ms)
    ✓ should return false for [0, 1, 2, 3]
    ✓ should return false for [Function func]
    ✓ should return false for [Function generatorFunc] (1ms)
    ✓ should return false if an imposter!
    ✓ should return false if an async generator
    ✓ should return true if a generator (1ms)
```